dipy/reconst/multi_voxel.py

```diff
@@ -5,6 +5,224 @@
 from dipy.core.ndindex import ndindex
 from dipy.reconst.quick_squash import quick_squash as _squash
 from dipy.reconst.base import ReconstFit
+from multiprocessing import cpu_count, Pool, Array
+import ctypes
+import warnings
+import itertools
+
+
+_ctypes_to_numpy = {
+    ctypes.c_char: np.dtype(np.uint8),
+    ctypes.c_wchar: np.dtype(np.int16),
+    ctypes.c_byte: np.dtype(np.int8),
+    ctypes.c_ubyte: np.dtype(np.uint8),
+    ctypes.c_short: np.dtype(np.int16),
+    ctypes.c_ushort: np.dtype(np.uint16),
+    ctypes.c_int: np.dtype(np.int32),
+    ctypes.c_uint: np.dtype(np.uint32),
+    ctypes.c_long: np.dtype(np.int64),
+    ctypes.c_ulong: np.dtype(np.uint64),
+    ctypes.c_float: np.dtype(np.float32),
+    ctypes.c_double: np.dtype(np.float64),
+    ctypes.c_void_p: np.dtype(object)}
+
+
+_numpy_to_ctypes = dict(zip(_ctypes_to_numpy.values(),
+                            _ctypes_to_numpy.keys()))
+
+global shared_arr
+
+
+def shm_as_ndarray(mp_array, shape=None):
```

Comment:
I wonder about the naming here: there's nothing specific to spherical harmonics here, is there? Or does "shm" stand for something else here?

Reply:
right, bad naming, `shm` for `shared multiprocessing array`. I will changed that